go/vt/vtgate/executor.go

```diff
@@ -1533,7 +1554,7 @@ func (e *Executor) initVConfig(warnOnShardedOnly bool, pv plancontext.PlannerVer
 
 	e.vConfig = econtext.VCursorConfig{
 		Collation:         connCollation,
-		DefaultTabletType: defaultTabletType,
+		DefaultTabletType: topodatapb.TabletType(defaultTabletType.Get()),
```

Comment:
we create this config only once. So it will not be dynamically updated on this config.